### DISCUSSION TITLE
May I ask if there is a tidal forcing generation API available now

### DISCUSSION THREAD

**Discussion by q1165600785** [COSIMA/mom6-regional-scripts#188] [General]
I want to use mom6 to simulate regional tides. Can I just input a tidal forcing

**Comment by ashjbarnes** [comment]
Hello! 
So body forcing tides (adding tidal forcing to the momentum equation) is already in MOM6, and just needs to be turned on. This doesn't require anything special for regional modelling. If you want to add boundary tides (so water sloshing in and out of the domain), which is very important for smaller regional models, you'll need to create boundary forcing segments specifically for the tides. 

Doing the latter in this package is currently a work in progress. I've got a hacky way of putting tides in using code modified / borrowed from [this really helpful repository](https://github.com/jsimkins2/nwa25/blob/main/setup/tides/write_tpxo.py) but we haven't yet written this into our package. This will hopefully be coming in the next couple of months. In the meantime, what you can do is use the code from this North West Atlantic repository to create your tide boundary files from TPXO, and then use the rest of this package to generate all of your other boundary files. My experiments [here](https://github.com/ashjbarnes/tasman-tides/tree/main/rundirs/full-20) give an example of MOM_input files with tides turned on at the boundary.

**Reply by q1165600785** [reply]
I'm very sorry for replying to you only on site. I was researching the ADCirc model a while ago, and what I wanted to do was to use TPOX tides to add initial water level conditions to the regional boundary, and then start simulating the region. Do I need to turn on tidal forcing? Do I just need to add boundary tidal conditions according to the above code to start my regional tidal simulation

**Reply by q1165600785** [reply]
Doctor, I would also like to add that I only focus on the height of the tide, so is it sufficient to only provide the tidal height as the boundary forcing condition, without the need for flow velocity

**Reply by ashjbarnes** [reply]
Hello, have a look at the huge pull request which includes the addition of tides. There's also an example jupyter notebook using this fork here which will be merged with the main repo at some point:

https://github.com/CROCODILE-CESM/regional-casegen/blob/main/demo.ipynb

**Reply by q1165600785** [reply]
Thank you for your answer, Doctor. Do I need to install which branch to use this function, Doctor? Which branch is it, or can I install the distribution package to use this function, Doctor

**Reply by ashjbarnes** [reply]
Easiest if you use the CROCODILE-CESM fork of the repository for now, as this is where the pull request is coming from. Note that this PR is still a draft and going through testing, so we can't guarantee that it's in good working order until it's finished and merged with the main COSIMA branch

**Reply by q1165600785** [reply]
Doctor, I looked at that demo and it seems that there is no usage method for inserting data into the TPOX ocean tide model. Momo6 forces data, and I only input the TPOX boundary water level height. Can I not input Gloris and EAR5 data

**Reply by ashjbarnes** [reply]
Hi, sorry I don't quite understand - what do you mean by inserting data into the TPOX model? The demo notebook linked above runs MOM6 forced by GLORYS, TPXO and reads in GEBCO bathymetry. @helenmacdonald is working on another example notebook too but I think it will look fairly similar

**Reply by q1165600785** [reply]
Doctor, thank you for your help. I have successfully generated the input file, but when I run it, I get this error: FATAL: MOM_domains_init: The product of the two components of layout, 80 80, is not the number of PEs used, 1.
Where did I go wrong? Doctor, can you help me?

**Reply by ashjbarnes** [reply]
Have you been able to get mom6 running on your computer in general? This package doesn’t help with getting mom6 itself set up - you need to make sure you can run the basic mom6 tests successfully with your executable, then you can try to run regional cases set up with this package 

This error means that’s you’re only running mom6 with a single cpu core, whereas you’ve asked for 80x80

**Reply by q1165600785** [reply]
Doctor, I'm very sorry for getting back to you so late. As I still have production projects to complete at work, my research project has been forced to come to a standstill,
Doctor, I installed mom6 into the Windows system's WSL and now I want to use a single core mom6 to run the program. In which file should I modify the configuration parameters and turn 80 × 80 into a single core format? Do I also need to regenerate the input file

**Reply by ashjbarnes** [reply]
Hi, the package defaults to 10x10 if you run the `fre_tools` method. 80x80 suggests you're using the code in my Tasman Tides repository as a base? I'd suggest instead you use the demos included in the package. Try and get the basic demo running first without tides, then turn them on later.

The cpu layout is specified in the MOM_layout file. Please make sure you're using the branch of the package following on from our conversation on pull request #193 and follow the demos on that branch too
